ddtrace/internal/ci_visibility/encoder.py

```diff
@@ -67,24 +73,57 @@ def _convert_span(span, dd_origin):
         # type: (Span, str) -> Dict[str, Any]
         sp = JSONEncoderV2._span_to_dict(span)
         sp = JSONEncoderV2._normalize_span(sp)
-        sp["type"] = span.span_type
+        if dd_origin is not None:
+            sp["meta"].update({"_dd.origin": dd_origin})
+        sp["type"] = span.get_tag(EVENT_TYPE) or span.span_type
         sp["duration"] = span.duration_ns
         sp["meta"] = dict(sorted(span._meta.items()))
         sp["metrics"] = dict(sorted(span._metrics.items()))
-        sp["trace_id"] = int(sp.get("trace_id") or "1")
-        sp["parent_id"] = int(sp.get("parent_id") or "1")
-        sp["span_id"] = int(sp.get("span_id") or "1")
-        sp["test_suite_id"] = 1  # TODO: populate with real ID
-        sp["test_session_id"] = 1  # TODO: populate with real ID
-        if COVERAGE_TAG_NAME in sp["meta"]:
-            del sp["meta"][COVERAGE_TAG_NAME]
-        if dd_origin is not None:
-            sp["meta"].update({"_dd.origin": dd_origin})
+
+        sp = CIVisibilityEncoderV01._filter_ids(sp)
+
+        version = CIVisibilityEncoderV01.TEST_SUITE_EVENT_VERSION
+        if span.get_tag(EVENT_TYPE) == "test":
+            version = CIVisibilityEncoderV01.TEST_EVENT_VERSION
+
         if span.span_type == "test":
-            event_type = "test"
+            event_type = span.get_tag(EVENT_TYPE)
         else:
             event_type = "span"
-        return {"version": CIVisibilityEncoderV01.TEST_EVENT_VERSION, "type": event_type, "content": sp}
+
+        return {"version": version, "type": event_type, "content": sp}
+
+    @staticmethod
+    def _filter_ids(sp):
+        """
+        Remove trace/span/parent IDs if non-test event, move session/module/suite IDs from meta to outer content layer.
+        """
+        if sp["meta"].get(EVENT_TYPE) in [SESSION_TYPE, MODULE_TYPE, SUITE_TYPE]:
+            del sp["trace_id"]
+            del sp["span_id"]
+            del sp["parent_id"]
+        else:
+            sp["trace_id"] = int(sp.get("trace_id") or "0")
+            sp["parent_id"] = int(sp.get("parent_id") or "0")
+            sp["span_id"] = int(sp.get("span_id") or "0")
+        if sp["meta"].get(EVENT_TYPE) in [SESSION_TYPE, MODULE_TYPE, SUITE_TYPE, SpanTypes.TEST]:
+            test_session_id = sp["meta"].get("test_session_id")
+            if test_session_id:
+                sp["test_session_id"] = int(test_session_id)
```

Comment:
Using constants for these dictionary keys would make this code more future-proof.